analyses/cms-open-data-ttbar/statistical_interference/rebinning_tool.py

```diff
@@ -0,0 +1,156 @@
+import ROOT
+
+
+class RebinningTool:
+    """
+    Rebinning tool, which can be used for rebinning and cutting edges
+    Save histograms to new .root file, which have structure as input one
+
+    set_xmin: set lower edge for all histograms
+    set_xmax: set upper edge for all histograms
+    set_rebin: set rebin coefficient
+
+    example of usage:
+
+    rebinning_tool = RebinningTool()
+    rebinning_tool.set_xmin(110) # if xmax was not set -> keep original one
+    rebinning_tool.set_rebin(2)
+    rebinning_tool.set_input_path("data/histograms.root")
+    rebinning_tool.set_output_path("data/temp_histos.root")
+    rebinning_tool.apply_rebinning()
+    """
+
+    def __init__(self):
+        self.fHistoBinLow = None
+        self.fHistoBinHigh = None
+        self.fRebin = None
+        self.fRebinning = False
+
+    def set_xmin(self, xmin):
+        self.fHistoBinLow = xmin
+
+    def set_xmax(self, xmax):
+        self.fHistoBinHigh = xmax
+
+    def set_rebin(self, rebin):
+        self.fRebinning = True
+        self.fRebin = rebin
+
+    def list_histograms(self, directory, path=""):
+        keys = directory.GetListOfKeys()
+
+        # Loop over all keys
+        for key in keys:
+            # Get the object associated with the key
+            obj = key.ReadObj()
+
+            # Construct the full path of the object
+
+            if path == "":
+                obj_path = obj.GetName()
+            else:
+                obj_path = path + "/" + obj.GetName()
```

Comment:
```suggestion
            obj_path = os.path.join(path, obj.GetName())
```
`os.path.join` already implements this logic